tests/cypress/tests/unit/dash_table_queries_test.ts

```diff
@@ -0,0 +1,107 @@
+}
+const getQuerySyntaxTree = (query: string): any => new QuerySyntaxTree(query);
+const getMultiColumnSyntaxTree = (query: string): any => new MultiColumnsSyntaxTree(query);
+const getSingleColumnSyntaxTree = (query: string): any => new SingleColumnSyntaxTree(query, {
+    id: 'a'
+});
+
+describe('Dash Table Queries', () => {
+    R.forEach(c => {
+        describe(c.name, () => {
+            describe('relational operator', () => {
+                describe('eq', () => {
+                    processCases(c.syntaxer, [
+                        { name: 'compares "1" to 1', query: `${c.hideOperand ? '' : '{a} '}eq 1`, target: { a: '1' }, valid: true, evaluate: true },
+                        { name: 'compares "1" to "1"', query: `${c.hideOperand ? '' : '{a} '}eq "1"`, target: { a: '1' }, valid: true, evaluate: true },
+                        { name: 'compares "1.0" to 1', query: `${c.hideOperand ? '' : '{a} '}eq 1`, target: { a: '1.0' }, valid: true, evaluate: true },
+                        { name: 'compares "1.0" to "1"', query: `${c.hideOperand ? '' : '{a} '}eq "1"`, target: { a: '1.0' }, valid: true, evaluate: true },
+                        { name: 'compares "1.1" to 1', query: `${c.hideOperand ? '' : '{a} '}eq 1`, target: { a: '1.1' }, valid: true, evaluate: false },
+                        { name: 'compares "1.1" to "1"', query: `${c.hideOperand ? '' : '{a} '}eq "1"`, target: { a: '1.1' }, valid: true, evaluate: false },
+                        { name: 'compares "1" to 0x1', query: `${c.hideOperand ? '' : '{a} '}eq 0x1`, target: { a: '1' }, valid: true, evaluate: true },
+                        { name: 'compares "1" to "0x1"', query: `${c.hideOperand ? '' : '{a} '}eq "0x1"`, target: { a: '1' }, valid: true, evaluate: true },
+
+                        { name: 'compares 1 to 1', query: `${c.hideOperand ? '' : '{a} '}eq 1`, target: { a: 1 }, valid: true, evaluate: true },
+                        { name: 'compares 1 to "1"', query: `${c.hideOperand ? '' : '{a} '}eq "1"`, target: { a: 1 }, valid: true, evaluate: true },
+                        { name: 'compares 1.1 to 1', query: `${c.hideOperand ? '' : '{a} '}eq 1`, target: { a: 1.1 }, valid: true, evaluate: false },
+                        { name: 'compares 1.1 to "1"', query: `${c.hideOperand ? '' : '{a} '}eq "1"`, target: { a: 1.1 }, valid: true, evaluate: false },
+                        { name: 'compares 1 to 0x1', query: `${c.hideOperand ? '' : '{a} '}eq 0x1`, target: { a: 1 }, valid: true, evaluate: true },
+                        { name: 'compares 1 to "0x1"', query: `${c.hideOperand ? '' : '{a} '}eq "0x1"`, target: { a: 1 }, valid: true, evaluate: true },
+
+                        { name: 'compares "x1" to 1', query: `${c.hideOperand ? '' : '{a} '}eq 1`, target: { a: 'x1' }, valid: true, evaluate: false },
+                        { name: 'compares "x1" to "1"', query: `${c.hideOperand ? '' : '{a} '}eq "1"`, target: { a: 'x1' }, valid: true, evaluate: false }
```

Comment:
Great tests 🎉 A couple more cases I'd like for my peace of mind:
For `eq`:
`eq "1.0"` & `{a: '1'}` -> `true`
`eq " 1 "` & `{a: '\t1\t'}` -> `true`
`eq 1` & `{a: '1x'}` -> `false` (and reversed `eq '1x'` & `{a: 1}`)

For `contains`:
`contains "b"` & `{a: 'abc'}` -> `true`
`contains " b"` & `{a: 'abc'}` -> `false`
`contains "b "` & `{a: 'abc'}` -> `false`
`contains " b"` & `{a: 'a bc'}` -> `true`
`contains "b "` & `{a: 'ab c'}` -> `true`

For `like` (or whatever we call it), just a couple more showing that trailing zeros in the filter don't get dropped:
`like "2005-01-01 00"` & `{a: '2005-01-01 01'}` -> `false`
`like "2005-01-01 00"` & `{a: '2005-01-01 00:01'}` -> `true`
`like "2005-01-01 00:00"` & `{a: '2005-01-01 00:01'}` -> `false`
and a couple of 2-digit years to and from 2-digit and 4-digit years

Reply:
✍️ Added